flytekit/extras/tensorflow/record.py

```diff
@@ -81,11 +81,14 @@ def to_literal(
             )
         )
         local_dir = ctx.file_access.get_random_local_directory()
-        remote_path = ctx.file_access.get_random_remote_path()
+        remote_path = ctx.file_access.join(
+            ctx.file_access.raw_output_prefix,
+            ctx.file_access.get_random_string(),
+        )
         local_path = os.path.join(local_dir, "0000.tfrecord")
         with tf.io.TFRecordWriter(local_path) as writer:
             writer.write(python_val.SerializeToString())
-        ctx.file_access.put_data(local_path, remote_path, is_multipart=False)
+        remote_path = ctx.file_access.put_raw_data(local_path)
```

Comment:
do we need to override here? I thought type_engine would modify it

Reply:
i think we are going to override everywhere. TypeEngine could modify, but i want to be sure, we should do it wherever we can